<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
<title>Tool Guide</title>
<link href="/assets/site.css" media="all" rel="stylesheet" type="text/css"/>
</head>
<body style="background-image: none;">
<h1>Tool Guide</h1>
<h2>Introduction</h2>
<p>
	This tool guide is an overview of the GenomeSpace functionality in the user interfaces of the GenomeSpace tools and data sources.  This is not meant to be comprehensive documentation for the given tools.  For more information on each tool, see <a href="/support/tools">the list of GenomeSpace tools and the help links provided on each tool's page</a>.</p>
<p>
	If you are having problems launching certain desktop tools, which depend on the Java Network Launch Protocol (JNLP), see <a href="http://www.genomespace.org/faq#FAQ2.2" target="_self">FAQ2.2</a> for solutions. These solutions were updated February of 2015, and should take you about five minutes to incorporate. They involve adjusting your browser pop-up blocker and system's Java security exception site list.</p>
<p>
	 </p>
<p>
	For other tool-related documentation, check out the following guides:</p>
<p>
<strong style="font-size: 18px;"><a href="http://genomespace.org/support/guides/loading-data/">Loading Data into GenomeSpace</a></strong></p>
<p>
<strong style="font-size: 18px;"><a href="http://genomespace.org/support/guides/moving-data/">Moving Data Between Tools viaGenomeSpace</a></strong></p>
<hr/>
<h2>ArrayExpress</h2>
<p>
	Project Website: <a href="http://www.ebi.ac.uk/arrayexpress/">http://www.ebi.ac.uk/arrayexpress/</a></p>
<p>
<strong>ArrayExpress </strong>is a repository of over 30,000 functional genomics experiments comprising nearly 1 million assays. Users can query and retrieve data in  a number of different formats including the <a href="http://fged.org/projects/miame/">MIAME</a> and <a href="http://fged.org/projects/minseqe/">MINSEQE</a> standards.</p>
<p>
	The ArrayExpress detail view for experiments provides a link you can use to send the data associated with that view (IDF, SDRF, and data files) directly to your GenomeSpace account.  To do this:</p>
<ol>
<li>
		Search in ArrayExpress for an experiment that contains data of interest to you.  For more information about searching the ArrayExpress experiment archive, see the <a href="http://www.ebi.ac.uk/fg/doc/help/ae_help.html">ArrayExpress Help</a>.</li>
<li>
		Click the <em>Send [experiment ID] data to GenomeSpace</em> link on that experiment's detail page.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_ae_gs.png" style="width: 650px; height: 423px; border-width: 1px; border-style: solid;"/></p>
<ol start="3">
<li>
		Select the files that you want to upload to your account on GenomeSpace and click <em>Upload</em>.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_arrayex_sendtogs.jpg?1353364210" style="width: 598px; height: 191px;"/></p>
<p style="margin-left: 40px;">
	The uploader provides the upload status of each file.  When the upload for each file is complete, a green checkmark appears next to the file name.  When all the files are uploaded, a message appears to tell you so and also provides a link to GenomeSpace.</p>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_arrayex_sendtogscomplete.jpg?1353364372" style="width: 598px; height: 198px;"/></p>
<p style="margin-left: 40px;">
	You will find that ArrayExpress has created an ArrayExpress folder in your GenomeSpace cloud storage, with a subfolder that contains the experiment data files.  Future ArrayExpress exports will go into the ArrayExpress folder.</p>
<p style="margin-left: 40px;">
	Note that ArrayExpress experiment data files are exported from ArrayExpress in MAGE-TAB format.</p>
<p style="margin-left: 40px;">
<img alt="" height="232" src="/uploaded/content_toolug_arrayex_folderings.jpg?1353364656" width="653"/></p>
<p>
	 </p>
<p>
	Read more documentation on how to <a href="http://genomespace.org/support/guides/loading-data/sections/load-data-from-arrayexpress-to-genomespace">load data from ArrayExpress into GenomeSpace</a>.</p>
<p>
	Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-arrayexpress-to-gitools">send data from ArrayExpress to GiTools using GenomeSpace</a>.</p>
<hr/>
<h2>cBioPortal</h2>
<p>
	The cBioPortal for Cancer Genomics provides visualization, analysis and download of large-scale cancer genomics data sets.</p>
<h2>
	Sending Files from cBioPortal to GenomeSpace</h2>
<p>
	The cBioPortal "Download Data" tab on the home page provides an option for users to select a published dataset and save the entire file or a subset back to their GenomeSpace account.</p>
<ol>
<li>
		Click the "Download Data" tab on the home page of the cBioPortal<br/>
<img alt="" src="/uploaded/content_cbio_download.png" style="width: 178px; height: 36px; border: 1px solid black;"/></li>
<li>
		Using the provided form, select your dataset by providing a cancer study, genomic profile, patient/case set and gene set<br/>
<img alt="" src="/uploaded/content_cbio_form.png" style="width: 650px;"/></li>
<li>
		Click the Send Data Matrix to GenomeSpace button at the bottom of the form to create your dataset<br/>
<img alt="" src="/uploaded/content_cbio_gs_button.png" style="width: 93px; height: 44px;"/></li>
<li>
		You will be asked to authenticate your GenomeSpace account if you are not currently signed in<br/>
<img alt="" src="/uploaded/content_cbio_gs_auth.png" style="width: 527px; height: 332px; border: 1px solid black;"/></li>
<li>
		Select a directory to save your file and click <strong>Submit</strong><br/>
<img alt="" src="/uploaded/content_cbio_gs_data_dir.png" style="width: 497px; height: 456px; border: 1px solid black;"/></li>
</ol>
<p>
	 </p>
<p>
	Read more documentation how to <a href="http://genomespace.org/support/guides/loading-data/sections/load-data-from-the-cbioportal-to-genomespace">load data from cBioPortal to GenomeSpace</a>.</p>
<hr/>
<h2>Cistrome </h2>
<p>
	Project Website: <a href="http://cistrome.org/ap/">http://cistrome.org/ap/</a></p>
<p>
	A cistrome is defined as the set of cis-acting targets of a trans-acting factor on a genome scale. The Cistrome project integrates ChIP-chip/seq data for a cistrome with standard analysis pipelines, while  providing users with a user-friendly, powerful, flexible web interface with which they can access the data and pipelines.<br/>
<br/>
	In addition to the standard Galaxy functions, Cistrome has 29 ChIP-chip- and ChIP-seq-specific tools in three major categories, from preliminary peak calling and correlation analyses, to downstream genome feature association, gene expression analyses, and motif discovery.</p>
<hr/>
<h2>Cytoscape</h2>
<p>
	Project Website: <a href="http://www.cytoscape.org">http://www.cytoscape.org</a></p>
<p>
	Cytoscape is an open-source bioinformatics software platform for visualizing molecular interaction networks and biological pathways, and integrating these networks with annotations, gene expression profiles, and other state data.   Cytoscape core distribution provides a basic set of features for data integration and visualization.  Additional features are available as plugins.</p>
<p>
	Cytoscape is a Java-based application that downloads to your local machine when you launch it from GenomeSpace.  Once it's downloaded to your machine (after a launch from GenomeSpace), you can start it locally (by opening the JNLP) or launch from GenomeSpace.  If you launch Cytoscape from the Cytoscape website, at this time it will not be the version that is linked to GenomeSpace.</p>
<p>
	Like several other applications in the GenomeSpace suite of tools, Cytoscape can be expanded with plug-ins.  You can download plug-ins for your Cytoscape by choosing <em>Plugins&gt;Manage Plugins</em>.</p>
<p>
	Cytoscape has several GenomeSpace menu items.</p>
<h2>
	Logging In</h2>
<p>
	When you launch Cytoscape from GenomeSpace, your login should be handled seamlessly.</p>
<p>
	If you start from your GenomeSpace-enabled JNLP of Cytoscape and want to log into GenomeSpace in order to access your Amazon cloud storage or other GenomeSpace tools, you will be asked to log in with your GenomeSpace username and password.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytogslogin.jpg?1334770210" style="width: 300px; height: 206px;"/></p>
<p>
	You can also <em>Reset Password</em> on your existing GenomeSpace account.</p>
<h2>
	Import Menu</h2>
<p>
	Clicking <em>File&gt;Import&gt;GenomeSpace</em> provides several options for importing files from GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoimportgs.jpg?1331149766" style="width: 600px; height: 389px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 597px; height: 164px;">
<tbody>
<tr>
<td>
				Load Network</td>
<td>
				Loads a network from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Load Network from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of a network from a delimited text file or single-sheet MS Excel workbook.</td>
</tr>
<tr>
<td>
				Load Attributes</td>
<td>
				Loads attributes (for node/edge/network) from your GenomeSpace cloud storage. The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load CyTable</td>
<td>
				Loads a file containing attributes for a CyNetwork from your GenomeSpace cloud storage.  The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load Attributes from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of delimited text or single-sheet MS Excel workbook containing attribute data tables that are not formatted into Cytoscape node or edge attribute file formats.</td>
</tr>
<tr>
<td>
				Load Session</td>
<td>
				Loads a CYS file from your GenomeSpace cloud storage.  A session file (CYS) contains all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles present in a given working state.</td>
</tr>
<tr>
<td>
				Load Ontology and Annotations</td>
<td>
				Loads an annotation/ontology file from your GenomeSpace cloud storage. Annotations in Cytoscape are stored as a set of ontologies (a set of controlled vocabulary terms that annotate the objects).The standard file formats used in Cytoscape are OBO and Gene Association.</td>
</tr>
</tbody>
</table>
<h2>
<br/>
	Export Menu</h2>
<p>
	Clicking <em>File&gt;Export&gt;GenomeSpace</em> provides options for exporting files to GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoexportgs.jpg?1331154773" style="width: 600px; height: 353px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 598px; height: 126px;">
<tbody>
<tr>
<td>
				Upload File</td>
<td>
				Uploads a file from your local machine to the root level of your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Session As</td>
<td>
				Saves a session file (CYS), containing all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles in your present working state to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Network As</td>
<td>
				Save a network to your GenomeSpace cloud storage.</td>
</tr>
</tbody>
</table>
<h2>
	GenomeSpace Menu</h2>
<p>
	Clicking <em>File&gt;GenomeSpace</em> provides several options for managing your files in your GenomeSpace cloud storage and using GenomeSpace to launch other applications.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytofilegs.jpg" style="width: 600px; height: 365px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 500px;">
<tbody>
<tr>
<td>
				Delete File in GenomeSpace</td>
<td>
				Delete a file from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Download File</td>
<td>
				Download a file in your GenomeSpace cloud storage to your local machine.</td>
</tr>
<tr>
<td>
				List Available Files in GenomeSpace</td>
<td>
				Navigate your GenomeSpace directories.</td>
</tr>
<tr>
<td>
				Login To GenomeSpace</td>
<td>
				Log into your GenomeSpace account.</td>
</tr>
<tr>
<td>
				Launch</td>
<td>
				Launch another GenomeSpace tool</td>
</tr>
</tbody>
</table>
<p>
	 </p>
<hr/>
<h2>Cytoscape 3</h2>
<p>
	Project Website: <a href="http://www.cytoscape.org">http://www.cytoscape.org</a></p>
<p>
	New Documentation for Cytoscape 3 is on its way... In the meantime, below is the help from the previous revision (Cytoscape 2.8)</p>
<p>
	Cytoscape is an open-source bioinformatics software platform for visualizing molecular interaction networks and biological pathways, and integrating these networks with annotations, gene expression profiles, and other state data.   Cytoscape core distribution provides a basic set of features for data integration and visualization.  Additional features are available as plugins.</p>
<p>
	Cytoscape is a Java-based application that downloads to your local machine when you launch it from GenomeSpace.  Once it's downloaded to your machine (after a launch from GenomeSpace), you can start it locally (by opening the JNLP) or launch from GenomeSpace.  If you launch Cytoscape from the Cytoscape website, at this time it will not be the version that is linked to GenomeSpace.</p>
<p>
	Like several other applications in the GenomeSpace suite of tools, Cytoscape can be expanded with plug-ins.  You can download plug-ins for your Cytoscape by choosing <em>Plugins&gt;Manage Plugins</em>.</p>
<p>
	Cytoscape has several GenomeSpace menu items.</p>
<h2>
	Logging In</h2>
<p>
	When you launch Cytoscape from GenomeSpace, your login should be handled seamlessly.</p>
<p>
	If you start from your GenomeSpace-enabled JNLP of Cytoscape and want to log into GenomeSpace in order to access your Amazon cloud storage or other GenomeSpace tools, you will be asked to log in with your GenomeSpace username and password.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytogslogin.jpg?1334770210" style="width: 300px; height: 206px;"/></p>
<p>
	You can also <em>Reset Password</em> on your existing GenomeSpace account.</p>
<h2>
	Import Menu</h2>
<p>
	Clicking <em>File&gt;Import&gt;GenomeSpace</em> provides several options for importing files from GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoimportgs.jpg?1331149766" style="width: 600px; height: 389px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 597px; height: 164px;">
<tbody>
<tr>
<td>
				Load Network</td>
<td>
				Loads a network from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Load Network from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of a network from a delimited text file or single-sheet MS Excel workbook.</td>
</tr>
<tr>
<td>
				Load Attributes</td>
<td>
				Loads attributes (for node/edge/network) from your GenomeSpace cloud storage. The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load CyTable</td>
<td>
				Loads a file containing attributes for a CyNetwork from your GenomeSpace cloud storage.  The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load Attributes from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of delimited text or single-sheet MS Excel workbook containing attribute data tables that are not formatted into Cytoscape node or edge attribute file formats.</td>
</tr>
<tr>
<td>
				Load Session</td>
<td>
				Loads a CYS file from your GenomeSpace cloud storage.  A session file (CYS) contains all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles present in a given working state.</td>
</tr>
<tr>
<td>
				Load Ontology and Annotations</td>
<td>
				Loads an annotation/ontology file from your GenomeSpace cloud storage. Annotations in Cytoscape are stored as a set of ontologies (a set of controlled vocabulary terms that annotate the objects).The standard file formats used in Cytoscape are OBO and Gene Association.</td>
</tr>
</tbody>
</table>
<h2>
<br/>
	Export Menu</h2>
<p>
	Clicking <em>File&gt;Export&gt;GenomeSpace</em> provides options for exporting files to GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoexportgs.jpg?1331154773" style="width: 600px; height: 353px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 598px; height: 126px;">
<tbody>
<tr>
<td>
				Upload File</td>
<td>
				Uploads a file from your local machine to the root level of your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Session As</td>
<td>
				Saves a session file (CYS), containing all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles in your present working state to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Network As</td>
<td>
				Save a network to your GenomeSpace cloud storage.</td>
</tr>
</tbody>
</table>
<h2>
	GenomeSpace Menu</h2>
<p>
	Clicking <em>File&gt;GenomeSpace</em> provides several options for managing your files in your GenomeSpace cloud storage and using GenomeSpace to launch other applications.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytofilegs.jpg" style="width: 600px; height: 365px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 500px;">
<tbody>
<tr>
<td>
				Delete File in GenomeSpace</td>
<td>
				Delete a file from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Download File</td>
<td>
				Download a file in your GenomeSpace cloud storage to your local machine.</td>
</tr>
<tr>
<td>
				List Available Files in GenomeSpace</td>
<td>
				Navigate your GenomeSpace directories.</td>
</tr>
<tr>
<td>
				Login To GenomeSpace</td>
<td>
				Log into your GenomeSpace account.</td>
</tr>
<tr>
<td>
				Launch</td>
<td>
				Launch another GenomeSpace tool</td>
</tr>
</tbody>
</table>
<p>
	 </p>
<hr/>
<h2>FireBrowse</h2>
<p>
<a href="http://firebrowse.org/" target="new">FireBrowse</a> is a companion portal to the Broad Institute GDAC Firehose analysis pipeline, and was developed to cull and analyze data generated by <a href="http://cancergenome.nih.gov/" target="new">The Cancer Genome Atlas (TCGA)</a>, which characterizes and identifies genomic patterns in human cancer models. Backed by a powerful compute infrastructure, programming interface, online reports and modern graphical tools, FireBrowse provides a simple yet capable means of visually and programmatically exploring one of the most comprehensive and deeply-characterized open cancer datasets in the world. FireBrowse provides access to a variety of cancer genomics data, such as clinical annotations, DNA copy number, miR, miRseq, mRNA and mRNAseq; as well as a comprehensive suite of more than 100 interdependent analyses of those data, including: correlations, clustering, and <a href="http://www.ncbi.nlm.nih.gov/pubmed/21527027" target="new">GISTIC</a> and <a href="http://www.ncbi.nlm.nih.gov/pubmed/23770567" target="new">MutSigCV</a>. FireBrowse is developed at the Broad Institute of MIT and Harvard.</p>
<p>
	More information about FireBrowse can be found using the following links:</p>
<ul>
<li>
<a href="http://firebrowse.org/" target="new">FireBrowse.org</a></li>
<li>
<a href="http://firebrowse.org/tutorial/FireBrowse-Tutorial.pdf" target="new">FireBrowse tutorial</a></li>
<li>
<a href="https://confluence.broadinstitute.org/display/GDAC/FAQ" target="new">FireBrowse FAQ</a></li>
<li>
<a href="https://confluence.broadinstitute.org/display/GDAC/Data+Usage+Policy" target="new">FireBrowse TCGA Data Use Policy and Publication Guidelines</a></li>
</ul>
<p>
	 </p>
<h2>
	Downloading data from FireBrowse to GenomeSpace</h2>
<p>
	When you launch FireBrowse from GenomeSpace, you will be automatically redirected to the FireBrowse front page (<a href="http://firebrowse.org/" target="new">http://firebrowse.org/</a>). On this page, you can access FireBrowse data by selecting a cohort.</p>
<p>
	To do this, you can <strong>(A)</strong> select a cohort from the drop-down menu; or, <strong>(B)</strong> simply click on the respective row of the samples barchart for that cohort. In the example below, we will be downloading Breast invasive carcinoma (BRCA) data. Therefore, we can either: (A) “Breast invasive carcinoma (BRCA)” from the drop-down menu; or (B) click on the BRCA barchart.</p>
<p>
<img alt="" src="/uploaded/content_fb1.png" style="width: 600px; height: 458px;"/></p>
<p>
	Next, select the data that you wish to download. FireBrowse collects data from a variety of analyses: clinical, copy number, miR, miRseq, mRNA, mRNAseq, mutation and pathway analyses.</p>
<p>
	To select a data type, either: <strong>(A)</strong> select the data type from the left-hand menu; or <strong>(B)</strong> click on the corresponding barchart. In our example, we will download SNP6 Copy Number variation data. We therefore either: (A) Click on the "CopyNumber Analyses" box; or (B) click on the "SNP6 CopyNum" barchart.</p>
<p>
<img alt="" src="/uploaded/content_fb2.png" style="width: 600px; height: 438px;"/></p>
<p>
	When you select a data type to download, this will prompt a data download interface. To send data to GenomeSpace, use the following method:</p>
<ol start="1" style="list-style-type: upper-alpha;">
<li>
		Click on the "Send To" button.</li>
<li>
		Select which data file(s) to download by clicking on the checkboxes next to the file(s).</li>
<li>
		Click the "GenomeSpace Upload" button.</li>
</ol>
<p>
<img alt="" src="/uploaded/content_fb3.png" style="width: 600px; height: 408px;"/></p>
<p>
	This will prompt you to choose a GenomeSpace directory to save the file(s) to. Once you have chosen a directory, you can click the "Submit" button. This will send the file(s) to your GenomeSpace directory as a <strong>.tar.gz</strong> file.</p>
<p>
	When you return to GenomeSpace, you will have to right-click on the file and choose "Expand Archive". This will unzip the .tar.gz file and create a new folder with the same name. The files you chose to download in Step B will be available in that folder.</p>
<hr/>
<h2>Galaxy</h2>
<p>
	Project Website: <a href="http://galaxyproject.org/">http://galaxyproject.org/</a></p>
<p>
	Galaxy is an open-source, scalable framework for tool integration that allows users to analyze multiple alignments, compare genomic annotations, and profile metagenomic samples, among many possible analyses; workflows allow the linking together of analyses.</p>
<p>
	Please note that during the GenomeSpace Beta, you will be connected with a Galaxy test server.</p>
<h2>
	Logging In</h2>
<p>
	When you launch Galaxy from GenomeSpace, you will see a login window.  <strong>Do not enter anything in the <em>Email address</em>, <em>Password</em>, or <em>OpenID URL</em> fields!</strong></p>
<p>
	In the final line under the OpenID login section ("Or, authenticate with your _____ account."), select <em>GenomeSpace</em> and click <em>Login</em>.</p>
<p>
<img alt="" src="/uploaded/content_toolug_galaxylogin.jpg?1332441605" style="width: 250px; height: 346px;"/></p>
<h2>
<br/>
	Galaxy Tools</h2>
<p>
	Galaxy has its own tools that perform different functions or analyses, and there are tools for interacting with GenomeSpace.</p>
<h3>
	Get Data</h3>
<p>
	To import data to Galaxy from GenomeSpace, click <em>Get Data</em> in the Tools menu.  Under that, select <em><span class="genomespace_file_browser_prod tool-link">GenomeSpace import</span> from file browser</em><em>.</em>  This opens your GenomeSpace directories.  The navigation is a little different here than it is in GenomeSpace, since there is no separate directories pane.  Double-click folders to open them, and double-click<em> ..(To Parent)</em> to return to the previous level of the folder structure.</p>
<p>
<img alt="" src="/uploaded/content_toolug_galaxygetdata.jpg?1335465598" style="width: 700px; height: 208px;"/></p>
<p>
	Select the checkboxes for the files you want to bring into Galaxy. You can then select a file format from the file's drop-down menu, if there is a converter.  Click the <em>Send to Galaxy</em> button to queue the files for import into Galaxy.</p>
<h3>
	Send Data</h3>
<p>
	To send data back to GenomeSpace from Galaxy, click <em>Send Data</em> in the Tools menu.  Under that, select <em>GenomeSpace Exporter</em>.</p>
<p>
<img alt="" src="/uploaded/content_toolug_galaxysenddata.jpg?1335465614" style="width: 700px; height: 375px;"/></p>
<ol>
<li>
		Select the dataset you want to export from the drop-down menu.</li>
<li>
		Select the target directory -- you can open the base directory by clicking the <img alt="" src="/uploaded/content_toolug_galaxyplusicon.jpg?1335282078" style="width: 18px; height: 16px;"/> so that you can select a subdirectory:</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_galaxyfilelist.jpg?1335282187" style="width: 145px; height: 138px;"/></p>
<ol start="3">
<li>
		You can enter a file name.  If you do not enter one, Galaxy will generate a file name automatically.</li>
<li>
		Click <em>Execute</em> to send the file to GenomeSpace.</li>
</ol>
<p> </p>
<p>Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-galaxy-to-igv">send data from Galaxy to IGV using GenomeSpace</a>.</p>
<hr/>
<h2>GenePattern</h2>
<p>
	Project Website: <a href="http://www.genepattern.org">http://www.genepattern.org</a></p>
<p>
	GenePattern is a powerful genomic analysis platform that provides access to more than 150 tools for gene expression analysis, proteomics, SNP analysis, flow cytometry, RNA-seq analysis, and common data processing tasks. A web-based interface provides easy access to these modules and allows for the creation of multi-step analysis pipelines that enable reproducible <em>in silico</em> research.</p>
<p>
	A GenomeSpace-enabled GenePattern installation has a number of elements in it that help move data files from GenomeSpace and analysis results to GenomeSpace.</p>
<h2>
	Logging In</h2>
<p>
	When you launch GenePattern from GenomeSpace, GenePattern allows you to log in using your GenomeSpace account.  You will see the GenePattern login screen, and you can click <em>Sign in Using GenomeSpace</em>.</p>
<p>
<img alt="" src="/uploaded/content_gp1.png" style="width: 576px; height: 200px;"/></p>
<p>
	If your GenomeSpace username is not recognized as being associated with any GenePattern username, GenePattern gives you the opportunity to either:</p>
<ul>
<li>
		associate your GenomeSpace account with your existing GenePattern account</li>
<li>
		generate a new GenePattern account that is associated with your GenomeSpace account</li>
</ul>
<h2>
	Sending Files to GenePattern from GenomeSpace</h2>
<p>
	GenePattern receives your files and offers you a number of options of how you would like it to handle those files.</p>
<p>
<img alt="" src="/uploaded/content_gp2.png" style="width: 572px; height: 400px; border-width: 1px; border-style: solid;"/></p>
<p>
	Each file can be saved to the Upload Directory by simply clicking <em>Submit</em> next to the <em>Save a Copy</em> field.  Or you can opt to send the file to a module.  If the file can be converted by one of the GenomeSpace conversion scripts, you can opt for a conversion as you send it to the module of your choice.</p>
<h2>
	GenomeSpace Tab</h2>
<p>
	Within the GenePattern UI, the right side of the window is occupied with several tabs that provide access to recently run jobs, uploaded files, and, in GenomeSpace-enabled installations, your GenomeSpace directories.  You can navigate these directories.</p>
<p>
<img alt="" src="/uploaded/content_gp3.png" style="width: 343px; height: 250px;"/></p>
<p>
	To use one of your GenomeSpace files in a module, drag and drop the name of the file on the field box.</p>
<p>
	 </p>
<h2>
	Save To GenomeSpace</h2>
<p>
	Files in the<em> Job Results</em> and <em>Uploads</em> tabs, as well as on the Job Results page, can be sent directly to GenomeSpace. Click the file name to generate a navigation menu for the file.</p>
<p>
<img alt="" src="/uploaded/content_gp5_2.png" style="width: 643px; height: 500px;"/></p>
<p>
	Then, in the drop-down menu, select the location in your GenomeSpace cloud storage where you want to save the file and click <em>Save</em>.</p>
<p>
<img alt="" src="/uploaded/content_gp6.png" style="width: 506px; height: 230px;"/></p>
<h2>
	GenomeSpace Menu</h2>
<p>
	There is also a menu of GenomeSpace functionality, allowing you to log out of GenomeSpace, open GenomeSpace, or go to the GenomeSpace web site.</p>
<p>
<img alt="" src="/uploaded/content_gp4.png" style="width: 500px; height: 129px;"/></p>
<p>
	 </p>
<p>
	Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-genepattern-to-igv">send data from GenePattern to IGV using GenomeSpace</a>.</p>
<p>
	Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-genomica-to-genepattern">send data from Genomica to GenePattern using GenomeSpace</a>.</p>
<hr/>
<h2>Genomica</h2>
<p>
	Project Website: <a href="http://genomica.weizmann.ac.il/">http://genomica.weizmann.ac.il/</a></p>
<p>
	Genomica is an analysis and visualization tool for genomic data that can integrate gene expression data, DNA sequence data, and gene and experiment annotation information.</p>
<p>
	Genomica is a Java-based application that downloads to your local machine when you launch it from GenomeSpace.  Once it's downloaded to your machine (after a launch from GenomeSpace), you can start it locally (by opening the JNLP) or launch from GenomeSpace.  If you launch Genomica from the Genomica website, at this time it will not be the version that is linked to GenomeSpace.</p>
<h2>
	Logging In</h2>
<p>
	When you launch Genomica from GenomeSpace, your login should be handled seamlessly.</p>
<p>
	If you start from your GenomeSpace-enabled JNLP of Genomica and want to log into GenomeSpace in order to access your cloud storage or other GenomeSpace tools, you need to select <em>GenomeSpace&gt;Login</em>.  You will be asked to log in with your GenomeSpace username and password.</p>
<h2>
	Menus</h2>
<p>
	Genomica has two menus containing GenomeSpace items.</p>
<h3>
	GenomeSpace Menu</h3>
<p>
	The GenomeSpace menu in Genomica allows you to interact with your GenomeSpace cloud storage in a number of ways.</p>
<p>
<img alt="" src="/uploaded/content_toolug_genomicagsmenu.jpg?1334947603" style="width: 325px; height: 240px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 605px; height: 231px;">
<tbody>
<tr>
<td>
				Login GenomeSpace/Logout GenomeSpace</td>
<td>
				Signs into or out of GenomeSpace.</td>
</tr>
<tr>
<td>
				Open From GenomeSpace</td>
<td>
<p>
					Opens the GenomeSpace file browser to enable you to open a file from your GenomeSpace cloud storage into Genomica.</p>
<p>
<img alt="" src="/uploaded/content_toolug_genomicagsbrowser.jpg?1331928311" style="width: 400px; height: 323px;"/></p>
</td>
</tr>
<tr>
<td>
				Save To GenomeSpace</td>
<td>
				Saves a file from Genomica to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Export Network To GenomeSpace</td>
<td>
				Export a network file from Genomica to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Launch GenomeSpace Tool</td>
<td>
<p>
					Allows you to launch another GenomeSpace-enabled tool from Genomica.</p>
<p>
<img alt="" src="/uploaded/content_toolug_genomicatoollaunch.jpg?1331929174" style="width: 350px; height: 210px;"/></p>
<p>
					If you click <em>Launch with GS File</em>, the GenomeSpace file browser opens to allow you to select a file to use in launching your selected tool.</p>
</td>
</tr>
<tr>
<td>
				Upload Local File To GenomeSpace</td>
<td>
				Uploads a file from your local machine to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Download GenomeSpace File to Local File</td>
<td>
				Downloads a file from your GenomeSpace cloud storage to your local machine.</td>
</tr>
<tr>
<td>
				Delete From GenomeSpace</td>
<td>
				Deletes a file from your GenomeSpace cloud storage.</td>
</tr>
</tbody>
</table>
<h3>
	Sets Menu</h3>
<p>
	The <em>Sets</em> menu contains two GenomeSpace options.</p>
<p>
<img alt="" src="/uploaded/content_toolug_genomicasetsmenu.jpg?1331922478" style="width: 400px; height: 186px;"/></p>
<p>
	With the <em>Sets</em> menu, you can load either Gene Sets or Experiment Sets from GenomeSpace.</p>
<h2>
	In the User Interface</h2>
<p>
	Select <em>Algorithms&gt;Create a Module Network...</em>  Under the <em>Regulation</em> tab is a <em>GenomeSpace Load</em> button that enables you to load a list of candidate regulator genes for the analysis.</p>
<p>
<img alt="" src="/uploaded/content_toolug_genomica_lmn_gsload.jpg?1334946901" style="width: 600px; height: 295px;"/></p>
<p> </p>
<p>Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-genomica-to-genepattern">send data from Genomica to GenePattern using GenomeSpace</a>.</p>
<hr/>
<h2>geWorkbench</h2>
<p>
	Project Website: <a href="http://wiki.c2b2.columbia.edu/workbench/index.php/Overview">http://wiki.c2b2.columbia.edu/workbench/index.php/Overview</a></p>
<p>
	geWorkbench is an open-source bioinformatics platform that offers a comprehensive and extensible collection of tools for the management, analysis, visualization, and annotation of biomedical data. Many kinds of analysis are supported.  For microarrays, there are tools for filtering and normalization, basic statistical analyses, clustering, network reverse engineering, as well as many common visualization tools. For sequence data, there are routines such as BLAST, pattern detection, transcription factor mapping, and syntenic region analysis. geWorkbench provides access to a variety of external data sources, including microarray gene expression repositories (caArray), BLAST, gene annotation pages (via caBIO),protein and DNA sequence retrieval, and pathway diagrams (BioCarta), as well as acting a gateway to several computational services currently hosted on Columbia servers and clusters, including Pattern Discovery, Pudge (protein structure modeling), and SkyBase (database of molecular models).</p>
<h2>
	Logging In</h2>
<p>
	When you launch geWorkbench from GenomeSpace, you will se the Launch geWorkbench dialog.</p>
<p>
<img alt="" src="/uploaded/content_gew_installprompt.jpg?1349455889" style="width: 421px; height: 254px;"/></p>
<p>
<strong>If you have installed geWorkbench in the default location previously, click <em>OK</em>.</strong>  This will launch geWorkbench.</p>
<p>
	If you have never used geWorkbench before, or have not used the current version of geWorkbench, you need to install it.</p>
<h3>
	Installing geWorkbench</h3>
<p>
	If you have never used geWorkbench before, you can use the prompt you receive when you launch it from GenomeSpace to install it.  If you have used it and installed it in a non-standard directory, you can point GenomeSpace to that location in the prompt dialog.</p>
<p>
	To install geWorkbench:</p>
<ol>
<li>
		Click the link in the "please click <u>here</u> to install it" line and you will be taken to a page of installers. </li>
<li>
		Select and download the appropriate installer for your machine.</li>
<li>
		Expand the ZIP archive containing the installer.</li>
<li>
		Run the installer and go through the installation wizard.  It is recommended that you install geWorkbench in the default location; if you do not, you will have to enter the nonstandard location path in the prompt window every time you launch the tool.</li>
<li>
		Go back to the prompt dialog and click <em>OK</em>. This should complete the launch of geWorkbench you began in GenomeSpace.</li>
</ol>
<h2>
	Enabling GenomeSpace Within geWorkbench</h2>
<p>
	To access GenomeSpace in geWorkbench, enable the GenomeSpace component using the Component Configuration Manager.</p>
<table border="0" cellpadding="1" cellspacing="1" height="216" width="705">
<tbody>
<tr>
<td>
<ol>
<li>
						Select <em>Tools&gt; Component Configuration</em>.</li>
</ol>
</td>
<td>
<img alt="" src="/uploaded/content_gew_selectcompconfig.jpg?1349466184" style="width: 423px; height: 162px;"/></td>
</tr>
<tr>
<td>
<ol start="2">
<li>
						In the Component Configuration Manager dialog, select the <em>Genome Space</em> checkbox.</li>
<li>
						Click <em>Apply.</em></li>
<li>
						Click <em>Close</em>.</li>
</ol>
</td>
<td>
<img alt="" src="/uploaded/content_gew_compconfig.jpg?1349971336" style="width: 575px; height: 380px;"/></td>
</tr>
</tbody>
</table>
<h2>
	Using GenomeSpace in geWorkbench</h2>
<p>
	When you click <em>Workspace</em> in the upper left corner of the screen, the GenomeSpace pane will appear to the right.</p>
<p>
<img alt="" src="/uploaded/content_gew_workspacegs.jpg?1349468413" style="width: 504px; height: 411px;"/></p>
<p>
	Log in by clicking the <em>Login</em> button next to the GenomeSpace Directories header, and now the GenomeSpace pane will display your directories.</p>
<p>
<img alt="" src="/uploaded/content_gew_loggedin.jpg?1349469679" style="width: 371px; height: 223px;"/></p>
<p>
	You can work with your files within GenomeSpace:</p>
<p>
<img alt="" src="/uploaded/content_gew_gsdiraxns.jpg?1349469952" style="width: 488px; height: 312px;"/></p>
<h3>
	Moving Files From GenomeSpace to geWorkbench</h3>
<p>
	To move one of your GenomeSpace files into geWorkbench:</p>
<ol>
<li>
		If you do not already have a project in geWorkbench, right-click on <em>Workspace</em> and select <em>New Project</em> in the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewnewproj.jpg?1351713813" style="width: 550px; height: 286px;"/></p>
<ol start="2">
<li>
		Select a project.</li>
<li>
		Right-click the file you want to use in geWorkbench and select <em>Download</em> in the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewdownload.jpg?1351713961" style="width: 549px; height: 279px;"/></p>
<p style="margin-left: 40px;">
<strong>Note</strong> that if you download a file that is not in a geWorkbench file format, GenomeSpace will convert it if there is a converter available.</p>
<h3>
	Moving Files from geWorkbench to GenomeSpace</h3>
<p>
	To move files from geWorkbench to GenomeSpace:</p>
<ol>
<li>
		Click the <em>GenomeSpace</em> tab.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" height="313" src="/uploaded/content_toolug_gewgstab.jpg?1351795258" width="670"/></p>
<ol start="2">
<li>
		Right-click the file you want to move to GenomeSpace and select <em>Upload to GenomeSpace</em> from the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewupload.jpg?1351795545" style="width: 549px; height: 267px;"/></p>
<ol start="3">
<li>
		Select the GenomeSpace directory to which you want to upload the geWorkbench file.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewselectgsfolder.jpg?1351715808" style="width: 450px; height: 234px;"/></p>
<ol start="4">
<li>
		Click <em>OK</em>.</li>
</ol>
<hr/>
<h2>Gitools</h2>
<p>
	Project Website: <a href="http://www.gitools.org/">http://www.gitools.org/</a></p>
<p>
	Gitools is a framework for analysis and visualization of genomic data using interactive heatmaps that also allows data to be imported from various sources, including Biomart, Ensembl, IntOGen, and GenomeSpace.<br/>
<br/>
	In the Gitools interactive heatmaps, every cell is associated with one or several values, and users can visualize them and navigate among them.  Multiple annotation labels can be added to rows and columns. A number of actions can be performed within the heatmap that help to explore and interpret the data effectively, such as search, filter by value or label, cluster the heat-map, sort rows and columns by different criteria, and more. There are many editable features that make the heatmaps thoroughly customizable, such as color scales, size of cells, grid lines, font size, and color and type for labels.<br/>
<br/>
	It is possible to load multidimensional data genomic sets into the heatmaps.  For example, a cell in a heatmap that represents a gene in a tumour sample can represent the expression level, mutation status, copy number alterations, and methylation changes. With these properties complemented by a set of analyses (such as correlations, overlaps, enrichment, or group comparison), the exploration of large multidimensional genomics datasets becomes very effective.</p>
<h2>
	Importing Data From GenomeSpace</h2>
<p>
	To open a data file from GenomeSpace, you can either click the GenomeSpace icon on the Gitools welcome page:</p>
<p>
<img alt="" src="/uploaded/content_gitools_open.jpg?1359565739" style="width: 399px; height: 191px;"/></p>
<p>
	Or by selecting <em>File&gt;Open&gt;Heatmap from Genomespace...</em></p>
<p>
<em><img alt="" src="/uploaded/content_gitools_file_open.jpg?1359565784" style="width: 399px; height: 235px;"/></em></p>
<p>
	 </p>
<p>Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-arrayexpress-to-gitools">send data from ArrayExpress to GiTools using GenomeSpace</a>.</p>
<hr/>
<h2>Integrative Genomics Viewer (IGV)</h2>
<p>
	Project Website: <a href="http://www.broadinstitute.org/igv/">http://www.broadinstitute.org/igv/</a></p>
<h2>
	Logging In</h2>
<p>
	When you launch IGV from GenomeSpace, your login should be handled seamlessly.</p>
<h2>
	Sending Files to IGV From GenomeSpace</h2>
<p>
	When you <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-genomespace-to-a-tool">send to a data file to IGV from GenomeSpace</a>, IGV opens up with the file in the viewer, as shown below.  You can then rearrange the data tracks by dragging and dropping them where you want to view them, or otherwise manipulate the view.</p>
<p>
<img alt="" src="/uploaded/content_toolug_igvmain.jpg?1330965269" style="width: 640px; height: 254px;"/></p>
<h2>
	GenomeSpace Menu</h2>
<p>
	Within the IGV UI, there is a menu for GenomeSpace functionality:</p>
<p>
<img alt="" src="/uploaded/content_toolug_igvgsmenu.jpg?1335283515" style="width: 250px; height: 109px;"/></p>
<h3>
	Load from GenomeSpace</h3>
<p>
	This opens up a dialog box that allows you to load one data file from your GenomeSpace directories in the cloud.</p>
<p>
<img alt="" src="/uploaded/content_toolug_igvloadfrom.jpg?1330970028" style="width: 400px; height: 268px;"/></p>
<p>
	Navigate through the directories and select the file you want to load.  Click <em>Open</em>.</p>
<h3>
	Save Session to GenomeSpace</h3>
<p>
	A <em>session</em> in IGV is an XML snapshot of the current state of IGV (e.g., the genome that is selected, the data file(s) that is/are loaded into the viewer, the regions of interest that are defined).  You can save an IGV session locally normally, but with the GenomeSpace menu, you can save an IGV session to your GenomeSpace cloud storage.</p>
<p>
<img alt="" src="/uploaded/content_toolug_igvsaveas.jpg?1330970273" style="width: 400px; height: 282px;"/></p>
<h3>
	Load Session From GenomeSpace</h3>
<p>
	Loading a session file from GenomeSpace is similar to loading a data file, and uses a very similar dialog box.</p>
<p>
	 </p>
<p>
	Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-galaxy-to-igv">send data from Galaxy to IGV using GenomeSpace</a>.</p>
<p>
	Learn how to <a href="http://genomespace.org/support/guides/moving-data/sections/send-data-from-genepattern-to-igv">send data from GenePattern to IGV using GenomeSpace</a>.</p>
<hr/>
<h2>ISAcreator</h2>
<p>
	Project Website: <a href="http://isa-tools.org/">http://isa-tools.org/</a></p>
<p>
	The open source ISA metadata tracking tools help to manage an increasingly diverse set of life science, environmental, and biomedical experiments that employ one or a combination of technologies.</p>
<h2>
	Logging In</h2>
<p>
	When you launch ISAcreator from GenomeSpace, the Java application launches and gives you three options, including GenomeSpace:</p>
<p>
<img alt="" src="/uploaded/content_isa_open.jpg" style="width: 614px; height: 186px;"/></p>
<p>
	Click the <em>GenomeSpace</em> button and you will be asked to log in with your GenomeSpace username and password. </p>
<p>
<img alt="" src="/uploaded/content_isa_login.jpg" style="width: 550px; height: 385px;"/></p>
<p>
	Once you log in, you'll be asked to select a configuration to load.  A configuration is a template for describing and curating experiments.</p>
<p>
<img alt="" src="/uploaded/content_isa_configload.jpg" style="width: 549px; height: 384px;"/></p>
<h2>
	Opening Files from GenomeSpace</h2>
<p>
	When you have logged in and have your configuration loaded, you can then load ISA-TAB datasets from GenomeSpace.  From the main menu, select <em>load an existing isatab file</em>.</p>
<p>
<img alt="" height="241" src="/uploaded/content_isa_loadfromgs.jpg" width="299"/></p>
<p>
	Click <em>search genomespace</em>.</p>
<p>
<img alt="" height="294" src="/uploaded/content_isa_searchgs.jpg" width="300"/></p>
<p>
	In the GenomeSpace dialog, browse to the directory containing your ISA-TAB datasets, select it, and click <em>load selected</em>.</p>
<p>
<img alt="" height="244" src="/uploaded/content_isa_browseload.jpg" width="300"/></p>
<p>
	This loads the ISA-TAB files into your investigation.</p>
<p>
<img alt="" src="/uploaded/content_isa_tabsetloaded.jpg" style="width: 649px; height: 455px;"/></p>
<h2>
	Saving Files to GenomeSpace</h2>
<p>
	You can save your configuration and ISA-TAB files to GenomeSpace, using either <em>File&gt;save in GenomeSpace</em> or <em>File&gt;</em><em>save in GenomeSpace as</em>...</p>
<p>
<img alt="" src="/uploaded/content_isa_savetogs.jpg" style="width: 250px; height: 255px;"/></p>
<p>
	For more information on defining your studies, see the <a href="http://isatab.sourceforge.net/isahelp/ch04.html">ISAcreator User Manual</a>.</p>
<p>
	 </p>
<hr/>
<h2>MSigDB Online Tools</h2>
<p>
	Project Website: <a href="http://www.broadinstitute.org/msigdb">http://www.broadinstitute.org/msigdb</a></p>
<h2>
	Logging In</h2>
<p>
	When you launch MSigDB from GenomeSpace, GenomeSpace tries to log into MSigDB with your email address.  If you log in successfully, you will be brought to the <em>Investigate Gene Sets</em> page.  <strong>Note</strong> that MSigDB is closely affiliated with the Gene Set Enrichment Analysis (GSEA) tool, and thus shares its website.  You are in the correct place, even though the logo in the corner says <em>GSEA</em>.</p>
<p>
<img alt="" height="386" src="/uploaded/content_msigdb_investgenesets.jpg" width="607"/></p>
<p>
	If the email address you use for GenomeSpace is not registered with MSigDB, you will be informed that the login failed.  Click the <em>click here</em> link to register as a new user, or log in with a registered email address.</p>
<p>
<img alt="" src="/uploaded/content_msigdb_loginfailreg.jpg" style="width: 450px; height: 237px;"/></p>
<p>
	Even if the email addresses are different, as long as you are logged into GenomeSpace and MSigDB concurrently, they can send data to one another.</p>
<h2>
	Sending Data to the MSigDB Online Tools from GenomeSpace</h2>
<p>
	You can send a list of genes in text format from GenomeSpace to be used as input the MSigDB Online Tools.  The list can be one per line, or delimited by spaces, tabs, or commas.</p>
<p>
	From GenomeSpace, you can:</p>
<ul>
<li>
		left-click the file and select <em>MSigDB</em> from the drop-down list</li>
</ul>
<p style="margin-left: 40px;">
	OR</p>
<ul>
<li>
		select the file and click the <em>MSigDB</em> icon in the tool bar</li>
</ul>
<p>
	Your gene list will populate the <em>Gene Identifiers</em> box on the Investigate Gene Sets page.</p>
<p>
	Note that you can also launch MSigDB by clicking its icon in the tool bar, then paste a list of genes into the <em>Gene Identifier</em> box.</p>
<p>
<img alt="" height="532" src="/uploaded/content_msigdb_investgenesets_genes.jpg" width="550"/></p>
<p>
	From here, select one or more collections in the Compute Overlaps list and click <em>compute overlaps</em>.</p>
<p>
	For more information on using the Compute Overlaps tool, see the <a href="http://www.broadinstitute.org/gsea/msigdb/help_annotations.jsp#overlap">Compute Overlaps help</a>.</p>
<h2>
	Save To GenomeSpace</h2>
<p>
	From the Compute Overlaps results page, you can click the <em>Save to: GenomeSpace</em> link to save your results to GenomeSpace.</p>
<p>
<img alt="" height="328" src="/uploaded/content_msigdb_computeoverlap_savetogs.jpg" width="500"/></p>
<p>
	In the dialog, browse to the directory you want to save the overlap file to and click <em>Submit</em>.</p>
<p>
<img alt="" height="335" src="/uploaded/content_msigdb_savetogs.jpg" width="356"/></p>
<p>
	 </p>
<hr/>
<h2>UCSC Table Browser</h2>
<h2>
	Sending Files from the UCSC Table Browser to GenomeSpace</h2>
<p>
	Project Website: <a href="http://genome.ucsc.edu/index.html">http://genome.ucsc.edu/index.html</a></p>
<p>
	The UCSC Table Browser is a valuable data source linked to GenomeSpace.  Since it is a data source, you cannot launch the UCSC Table Browser on a file in GenomeSpace.  You can only launch the UCSC Table Browser from its button in GenomeSpace, and then send files from the UCSC Table Browser to GenomeSpace.</p>
<p>
	The Table Browser allows you to retrieve data associated with a track in text format, to calculate intersections between tracks, and to retrieve DNA sequence covered by a track.  After you select the options for your output file, you can opt to send your output file to your GenomeSpace cloud storage.  For help using the Table Browser's options, see <a href="http://hgwdev-gs.cse.ucsc.edu/goldenPath/help/hgTablesHelp.html">the UCSC Table Browser documentation</a>.</p>
<p>
<img alt="" src="/uploaded/content_toolug_ucsc_gstablebrowser.jpg?1331845317" style="width: 700px; height: 359px;"/></p>
<p>
	To output your selected track to GenomeSpace, remember to select the checkbox for <em>GenomeSpace</em>. </p>
<p>
	Select an <em>output format</em>.  In the <em>output file</em> field, you need to specify the name of your output file.  There are two options for this:</p>
<ul>
<li>
		You can provide a file name, e.g., <span style="font-family: courier new,courier,monospace;">my_tracks</span>.  Providing a file name will save this file to the top level of your GenomeSpace directories (/Home/[username]).</li>
<li>
		You can provide a path.  Specifying a path will save the file to the specified directory under your GenomeSpace home directory. For example, specifying datasets/study1/my_tracks will save your file to the /Home/[username]/datasets/study1 directory.</li>
</ul>
<p style="margin-left: 40px;">
<strong>NOTE</strong>: You <strong>must</strong> add the correct extension to your file name. If, for instance, you select BED in the <em>output format </em>drop-down, you will need to add .bed to your output file name.  The UCSC Table Browser does not do this for you, and some of the GenomeSpace tools depend on the file extension to determine file type.</p>
<p>
	When you click <em>get output</em>, you will be asked to log into GenomeSpace, if you have not previously logged in via the UCSC Genome Browser.</p>
<p>
<img alt="" src="/uploaded/content_toolug_ucsclogin.jpg?1331845693" style="width: 275px; height: 170px;"/></p>
<p>
	After you log in (or if you previously logged in), you will see an uploading message.</p>
<p>
<img alt="" src="/uploaded/content_toolug_ucscuploading.jpg?1331845730" style="width: 250px; height: 58px;"/></p>
<p>
	Followed by a completion message.</p>
<p>
<img alt="" src="/uploaded/content_toolug_ucscuploadcomplete.jpg?1331845750" style="width: 250px; height: 132px;"/></p>
<p>
	Click <em>Back</em> to return to the Table Browser.</p>
<p>
	The file you sent to GenomeSpace should appear in your root directory.</p>
<p>
	 </p>
<p>
	Read more documentation on how to <a href="http://genomespace.org/support/guides/loading-data/sections/load-data-from-ucsc-genome-browser-to-genomespace">load data from UCSC Genome Browser to GenomeSpace</a>.</p>
<hr/>
<h2>Tool Term Glossary</h2>
<p>
<a name="GSE"></a><strong>GSE</strong> <strong>Identifier</strong>:  A series record links together a group of related samples, providing a focal point and description for the whole study. Series records may also contain tables describing extracted data, summary conclusions, or analyses. Each series record is assigned a unique and stable GEO accession number (GSExxx).</p>
<p>
<strong>Module (GenePattern)</strong>: A module is a discrete portion of code/functionality that is packaged with launch code, UI, and metadata, and can be run within the GenePattern user interface.</p>
<hr/>
<h2>Trinity</h2>
<p>
	The Trinity Cancer Transcriptome Analysis Toolkit (CTAT) aims to provide tools for leveraging RNA-Seq to gain insights into the biology of cancer transcriptomes. Bioinformatics tool support is provided for mutation detection, fusion transcript identification, and de novo transcript assembly. Trinity CTAT was developed at The Broad Institute and is hosted on the <a href="https://galaxy.ncgas-trinity.indiana.edu" target="new">Trinity CTAT Galaxy portal</a> by the <a href="http://ncgas.org/" target="new">National Center for Genome Analysis Support (NCGAS) at Indiana University</a>. CTAT is funded by the <a href="http://itcr.nci.nih.gov/" target="new">National Cancer Institute Informatics Technology for Cancer Research (NCI ITCR)</a> program.</p>
<p>
	More information about Trinity CTAT can be found using the following links:</p>
<ul>
<li>
<a href="https://github.com/NCIP/Trinity_CTAT/wiki" target="new">Trinity Cancer Transcriptome Analysis Toolkit Website</a></li>
<li>
<a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki" target="new">RNA-Seq De novo Assembly Using Trinity</a></li>
<li>
<a href="https://galaxy.ncgas-trinity.indiana.edu" target="new">Trinity CTAT Galaxy portal</a></li>
<li>
<a href="https://groups.google.com/forum/#!forum/trinity_ctat_users" target="new">Trinity CTAT users Google Group</a></li>
<li>
<a href="http://ncgas.org/galaxy_trinity_faq.php" target="new">NCGAS Trinity Galaxy FAQ</a></li>
</ul>
<p>
	 </p>
<h2>
	Creating an Account</h2>
<p>
	Before you can launch Trinity from GenomeSpace, you must first sign up for a Trinity account through the National Center for Genome Analysis Support (NCGAS). This can be accomplished by filling out the <a href="https://redcap.uits.iu.edu/surveys/?s=Edbef4bGGb" target="new">Trinity Galaxy Registration Form</a>.</p>
<p>
	Trinity Galaxy support by NCGAS is publically funded, so users are required to sign up and describe the nature of their work and their sources of funding. If you have any questions about this form, please contact the NCGAS team at <a href="mailto:help@ncgas.org">help@ncgas.org</a>.</p>
<p>
	Once you have submitted your registration form, you should receive an email from the NCGAS team with your account information and temporary password. This email should arrive within 24 hours of submitting your application.</p>
<p>
	 </p>
<h2>
	Connecting Your Account</h2>
<p>
	Once you have signed up for and activated your Trinity CTAT account, you can connect your Trinity account to GenomeSpace using the following method:</p>
<ol>
<li>
		On the main login screen, bypass the Trinity Login section and instead navigate to the OpenID Login section.</li>
<li>
		Under the OpenID URL box, choose "GenomeSpace" from the drop-down menu to authenticate with your GenomeSpace account.</li>
<li>
		Click "Login".<br/>
<img alt="" src="/uploaded/content_trinity1.png" style="border-width: 1px; border-style: solid; height: 635px; width: 600px;"/></li>
<li>
		Once you have successfully authenticated with OpenID, you will also need to associate a Galaxy account with this ID. To do this, enter your Trinity username and password again and click "Login" (see below). <strong>Do not create a new account</strong>.<br/>
<img alt="" src="/uploaded/content_trinity2.png" style="width: 600px; height: 320px; border-width: 1px; border-style: solid;"/></li>
</ol>
<p>
	Once you have connected your account, you can begin using Trinity normally.</p>
<p>
	 </p>
<h2>
	Trinity Tools</h2>
<p>
	Trinity encompasses a suite of tools hosted on a Galaxy server. There are three types of tools: (1) tools for interacting with GenomeSpace; (2) auxiliary tools that are common to Galaxy builds; and, (3) Trinity-specific tools which perform different functions or analyses.</p>
<h3>
	Get Data</h3>
<p>
	To import data to Trinity from GenomeSpace, click <em>Get Data</em> in the Tools menu and select <em><span class="genomespace_file_browser_prod tool-link">GenomeSpace import</span> from file browser</em><em>.</em>  This opens a view of your GenomeSpace directories.  The navigation is a little different here than it is in GenomeSpace, since there is no separate directories pane.  Double-click folders to open them, and double-click<em> ..(To Parent)</em> to return to the previous level of the folder structure.</p>
<p>
<img alt="" src="/uploaded/content_trinity3.png" style="width: 251px; height: 631px;"/></p>
<p>
	Select the checkboxes for the files you want to bring into Trinity. You can optionally select a file format from the file's drop-down menu, provided there is a converter.  Click the <em>Send to Galaxy</em> button to queue the files for import into Trinity.</p>
<h3>
	Send Data</h3>
<p>
	To send data back to GenomeSpace from Trinity, click <em>Send Data</em> in the Tools menu and select <em>GenomeSpace Exporter</em>.</p>
<p>
<img alt="" src="/uploaded/content_trinity4.png" style="width: 251px; height: 358px;"/></p>
<ol>
<li>
		Select the dataset you want to export from the drop-down menu.</li>
<li>
		Select the target directory – you can open enclosing directories by clicking the <img alt="" src="/uploaded/content_toolug_galaxyplusicon.jpg?1335282078" style="width: 18px; height: 16px;"/> button.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_galaxyfilelist.jpg?1335282187" style="width: 145px; height: 138px;"/></p>
<ol start="3">
<li>
		You can optionally enter a file name.  If you do not enter one, Trinity will generate a file name automatically.</li>
<li>
		Click <em>Execute</em> to send the file to GenomeSpace.</li>
</ol>
<p>
	 </p>
<h3>
	Other Galaxy Tools</h3>
<p>
	Trinity also has built-in Galaxy tools to help with basic file manipulation and other functions. These tools can by found by clicking on <em>Helper Tools</em> in the Tools menu.</p>
<p>
<img alt="" src="/uploaded/content_trinity5.png" style="width: 251px; height: 313px;"/></p>
<p>
	To learn about other Galaxy tools that can be found in Trinity, please visit the <a href="https://wiki.galaxyproject.org/" target="new">Galaxy Wiki Resource</a>.</p>
<div>
	 </div>
<div>
<h3>
		Trinity-Specific Tools</h3>
<p>
		The Trinity CTAT suite of tools is specific to this build of Galaxy, and is not currently deployed on the main Galaxy server which is also connected to GenomeSpace. Currently, CTAT supports tools and functions for completing analyses such as:</p>
<ul>
<li>
			De novo transcriptome assembly</li>
<li>
			Mutation detection</li>
<li>
			Fusion transcript identification</li>
<li>
			Other tools for manipulation of FASTA files</li>
</ul>
<p>
<img alt="" src="/uploaded/content_trinity6.png" style="width: 251px; height: 837px;"/></p>
<p>
		More information about the capabilities of these specific tools can be found on the <a href="https://github.com/NCIP/Trinity_CTAT/wiki" target="new">Trinity CTAT wiki page</a>. In the future, Trinity CTAT may also support lincRNA classification, foreign transcript detection, and single-cell cancer transcriptome analysis.</p>
</div>
<p>
	 </p>
<hr/>
</body>
</html>