<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">

<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
<title>GenomeSpace: 
	  	Tool Guide
  </title>
<link href="/assets/site.css" media="screen" rel="stylesheet" type="text/css"/>
<link href="/assets/demo_table_jui.css" media="screen" rel="stylesheet" type="text/css"/>
<link href="/assets/jquery-ui-1.8.4.custom.css" media="screen" rel="stylesheet" type="text/css"/>
<script src="/assets/application.js" type="text/javascript"></script>
<link href="/assets/favicon.ico" rel="shortcut icon" type="image/vnd.microsoft.icon"/>
<meta content="authenticity_token" name="csrf-param"/>
<meta content="fNuYKBXltbqFNXDASy8H8dGluG+t+6aCZSB5HhlqTY4=" name="csrf-token"/>
</head>
<body>
<div id="page_wrapper">
<div id="page_content">
<div id="logo"><a href="/"><img alt="Gslogo" src="/assets/gslogo.png"/></a></div>
<div id="login_menu"><a href="https://gsui.genomespace.org/jsui/gsui.html">User Login</a>
</div>
<ul id="nav">
<li id="what-is-genomespace"><a href="/what-is-genomespace">What is GenomeSpace?</a>
</li>
<li id="tools"><a href="/support/tools">Tools</a>
</li>
<li id="recipes"><a href="http://recipes.genomespace.org">Recipes</a></li>
<li id="documentation"><a href="/documentation">Documentation</a>
<ul>
<li><a href="/support/guides/user-guide">User Guide</a>
<p>Read the detailed guide to the user interface</p>
</li>
<li><a href="/support/guides/tool-guide">Tool Guide</a>
<p>Learn how to access GenomeSpace functionality for each Tool</p>
</li>
<li><a href="/faq">FAQ</a>
<p>See answers to frequently asked questions</p>
</li>
<li><a href="/presentations-and-videos">Presentations and Videos</a>
<p>Access talks by the GenomeSpace community</p>
</li>
<li><a href="/support/guides/end-to-end-analyses">End-to-End Analyses</a>
<p>Provides description for end-to-end analyses.</p>
</li>
</ul>
</li>
<li id="developers"><a href="/for-developers">Developers</a>
<ul>
<li><a href="/what-can-genomespace-do-for-tool-developers">What Can GenomeSpace Do For Tool Developers?</a>
<p></p>
</li>
<li><a href="/adopters">Technical Documentation</a>
<p></p>
</li>
<li><a href="/source-code">Source Code and Distributions</a>
<p></p>
</li>
<li><a href="/developer-support-and-feedback">Developer Support and Feedback</a>
<p></p>
</li>
</ul>
</li>
<li id="support"><a href="/support">Support</a>
</li>
<li id="about"><a href="/about">About</a>
</li>
<li class="search">
<div id="search_box">
<form accept-charset="UTF-8" action="/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="✓"/></div>
<input class="search" id="search" name="search" type="text"/>
<input class="searchBtn" type="submit"/>
</form> </div>
</li>
</ul>
<script type="text/javascript">
//<![CDATA[

		// grab page category and use to highlight correct menu item.  if none present, parse url to find where the user is.
		var category = "";
		if ( !category ){
			url_parts = $(location).attr('pathname').split("/");
			url_string = url_parts[1] + "/" + url_parts[2]
			if (url_string == "support/tools"){
				category = "tools"
			}
			else if (url_string == "support/guides"){
				url_parts[3] == "recipes" ? category = "recipes" : category = "documentation"
			}
			else if (url_parts[1] == "genomespace-recipes")
			{
                category = "gsrecipes"
            }
			else if (url_string == "support/api"){
				category = "developers"
			}
			else if ((url_string == "register/undefined") || (url_string == "request_invite/undefined")) {
				category = "what-is-genomespace"
			}
			else if (url_string == "/undefined") {
				category = ""
			}
			else if (url_string == "support/undefined") {
				category = "support"
			}
		}
		$("#" + category).addClass("current");


//]]>
</script>
<div id="content_panel">
<table width="100%"><tr>
<td id="toc_wrapper">
<div id="toc">
<h3>Tool Guide</h3>
<ol>
<li class="" id="guide_section_1">
<a href="/support/guides/tool-guide/sections/tool-guide-intro">Introduction</a>
</li>
<li class="" id="guide_section_2">
<a href="/support/guides/tool-guide/sections/arrayexpress-GS-capabilities">ArrayExpress</a>
</li>
<li class="" id="guide_section_3">
<a href="/support/guides/tool-guide/sections/cbioportal-GS-capabilities">cBioPortal</a>
</li>
<li class="" id="guide_section_4">
<a href="/support/guides/tool-guide/sections/cistrome-GS-capabilities">Cistrome </a>
</li>
<li class="" id="guide_section_5">
<a href="/support/guides/tool-guide/sections/cytoscape-GS-capabilities">Cytoscape</a>
</li>
<li class="" id="guide_section_6">
<a href="/support/guides/tool-guide/sections/cytoscape_3-GS-capabilities">Cytoscape 3</a>
</li>
<li class="" id="guide_section_7">
<a href="/support/guides/tool-guide/sections/firebrowse-GS-capabilities">FireBrowse</a>
</li>
<li class="" id="guide_section_8">
<a href="/support/guides/tool-guide/sections/galaxy-GS-capabilities">Galaxy</a>
</li>
<li class="" id="guide_section_9">
<a href="/support/guides/tool-guide/sections/genepattern-GS-capabilities">GenePattern</a>
</li>
<li class="" id="guide_section_10">
<a href="/support/guides/tool-guide/sections/genomica-GS-capabilities">Genomica</a>
</li>
<li class="" id="guide_section_11">
<a href="/support/guides/tool-guide/sections/geworkbench-GS-capabilities">geWorkbench</a>
</li>
<li class="" id="guide_section_12">
<a href="/support/guides/tool-guide/sections/gitools-GS-capabilities">Gitools</a>
</li>
<li class="" id="guide_section_13">
<a href="/support/guides/tool-guide/sections/igv-GS-capabilities">Integrative Genomics Viewer (IGV)</a>
</li>
<li class="" id="guide_section_14">
<a href="/support/guides/tool-guide/sections/isacreator">ISAcreator</a>
</li>
<li class="" id="guide_section_15">
<a href="/support/guides/tool-guide/sections/msigdb-online-tools">MSigDB Online Tools</a>
</li>
<li class="" id="guide_section_16">
<a href="/support/guides/tool-guide/sections/ucsc_genome_browser-GS-capabilities">UCSC Table Browser</a>
</li>
<li class="" id="guide_section_17">
<a href="/support/guides/tool-guide/sections/tool-term-glossary">Tool Term Glossary</a>
</li>
<li class="" id="guide_section_18">
<a href="/support/guides/tool-guide/sections/trinity-GS-capabilities">Trinity</a>
</li>
</ol>
</div>
<p class="center"><a href="/support/guides/tool-guide/print"><img alt="Print_icon" src="/assets/print_icon.gif" title="Printer Friendly View"/></a></p>
</td>
<script type="text/javascript">
//<![CDATA[

		$(document).ready(function() {
			var guide_section = "guide_section_11";
			$("#" + guide_section + "> a").css("font-weight", "bold");
			if ( 'Tool Guide' == "Recipes" ) {
				$.each($("#toc a.toggle_link"), function() {
					$(this).click();
					var image = $(this).children("img");
					$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
				});
			}
		});

//]]>
</script>
<td id="main_content">
<div id="notice"></div>
<div id="alert"></div>
<h1>geWorkbench
	</h1>
<p>
	Project Website: <a href="http://wiki.c2b2.columbia.edu/workbench/index.php/Overview">http://wiki.c2b2.columbia.edu/workbench/index.php/Overview</a></p>
<p>
	geWorkbench is an open-source bioinformatics platform that offers a comprehensive and extensible collection of tools for the management, analysis, visualization, and annotation of biomedical data. Many kinds of analysis are supported.  For microarrays, there are tools for filtering and normalization, basic statistical analyses, clustering, network reverse engineering, as well as many common visualization tools. For sequence data, there are routines such as BLAST, pattern detection, transcription factor mapping, and syntenic region analysis. geWorkbench provides access to a variety of external data sources, including microarray gene expression repositories (caArray), BLAST, gene annotation pages (via caBIO),protein and DNA sequence retrieval, and pathway diagrams (BioCarta), as well as acting a gateway to several computational services currently hosted on Columbia servers and clusters, including Pattern Discovery, Pudge (protein structure modeling), and SkyBase (database of molecular models).</p>
<h2>
	Logging In</h2>
<p>
	When you launch geWorkbench from GenomeSpace, you will se the Launch geWorkbench dialog.</p>
<p>
<img alt="" src="/uploaded/content_gew_installprompt.jpg" style="width: 421px; height: 254px;"/></p>
<p>
<strong>If you have installed geWorkbench in the default location previously, click <em>OK</em>.</strong>  This will launch geWorkbench.</p>
<p>
	If you have never used geWorkbench before, or have not used the current version of geWorkbench, you need to install it.</p>
<h3>
	Installing geWorkbench</h3>
<p>
	If you have never used geWorkbench before, you can use the prompt you receive when you launch it from GenomeSpace to install it.  If you have used it and installed it in a non-standard directory, you can point GenomeSpace to that location in the prompt dialog.</p>
<p>
	To install geWorkbench:</p>
<ol>
<li>
		Click the link in the "please click <u>here</u> to install it" line and you will be taken to a page of installers. </li>
<li>
		Select and download the appropriate installer for your machine.</li>
<li>
		Expand the ZIP archive containing the installer.</li>
<li>
		Run the installer and go through the installation wizard.  It is recommended that you install geWorkbench in the default location; if you do not, you will have to enter the nonstandard location path in the prompt window every time you launch the tool.</li>
<li>
		Go back to the prompt dialog and click <em>OK</em>. This should complete the launch of geWorkbench you began in GenomeSpace.</li>
</ol>
<h2>
	Enabling GenomeSpace Within geWorkbench</h2>
<p>
	To access GenomeSpace in geWorkbench, enable the GenomeSpace component using the Component Configuration Manager.</p>
<table border="0" cellpadding="1" cellspacing="1" height="216" width="705">
<tbody>
<tr>
<td>
<ol>
<li>
						Select <em>Tools&gt; Component Configuration</em>.</li>
</ol>
</td>
<td>
<img alt="" src="/uploaded/content_gew_selectcompconfig.jpg" style="width: 423px; height: 162px;"/></td>
</tr>
<tr>
<td>
<ol start="2">
<li>
						In the Component Configuration Manager dialog, select the <em>Genome Space</em> checkbox.</li>
<li>
						Click <em>Apply.</em></li>
<li>
						Click <em>Close</em>.</li>
</ol>
</td>
<td>
<img alt="" src="/uploaded/content_gew_compconfig.jpg" style="width: 575px; height: 380px;"/></td>
</tr>
</tbody>
</table>
<h2>
	Using GenomeSpace in geWorkbench</h2>
<p>
	When you click <em>Workspace</em> in the upper left corner of the screen, the GenomeSpace pane will appear to the right.</p>
<p>
<img alt="" src="/uploaded/content_gew_workspacegs.jpg" style="width: 504px; height: 411px;"/></p>
<p>
	Log in by clicking the <em>Login</em> button next to the GenomeSpace Directories header, and now the GenomeSpace pane will display your directories.</p>
<p>
<img alt="" src="/uploaded/content_gew_loggedin.jpg" style="width: 371px; height: 223px;"/></p>
<p>
	You can work with your files within GenomeSpace:</p>
<p>
<img alt="" src="/uploaded/content_gew_gsdiraxns.jpg" style="width: 488px; height: 312px;"/></p>
<h3>
	Moving Files From GenomeSpace to geWorkbench</h3>
<p>
	To move one of your GenomeSpace files into geWorkbench:</p>
<ol>
<li>
		If you do not already have a project in geWorkbench, right-click on <em>Workspace</em> and select <em>New Project</em> in the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewnewproj.jpg" style="width: 550px; height: 286px;"/></p>
<ol start="2">
<li>
		Select a project.</li>
<li>
		Right-click the file you want to use in geWorkbench and select <em>Download</em> in the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewdownload.jpg" style="width: 549px; height: 279px;"/></p>
<p style="margin-left: 40px;">
<strong>Note</strong> that if you download a file that is not in a geWorkbench file format, GenomeSpace will convert it if there is a converter available.</p>
<h3>
	Moving Files from geWorkbench to GenomeSpace</h3>
<p>
	To move files from geWorkbench to GenomeSpace:</p>
<ol>
<li>
		Click the <em>GenomeSpace</em> tab.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" height="313" src="/uploaded/content_toolug_gewgstab.jpg" width="670"/></p>
<ol start="2">
<li>
		Right-click the file you want to move to GenomeSpace and select <em>Upload to GenomeSpace</em> from the popup menu.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewupload.jpg" style="width: 549px; height: 267px;"/></p>
<ol start="3">
<li>
		Select the GenomeSpace directory to which you want to upload the geWorkbench file.</li>
</ol>
<p style="margin-left: 40px;">
<img alt="" src="/uploaded/content_toolug_gewselectgsfolder.jpg" style="width: 450px; height: 234px;"/></p>
<ol start="4">
<li>
		Click <em>OK</em>.</li>
</ol>
</td>
<tr><td colspan="2">
<p class="center">
<a href="/support/guides/tool-guide/sections/genomica-GS-capabilities">&lt;&lt; Genomica</a>
		| <a href="/support/guides/tool-guide/sections/tool-guide-intro">Up</a> |
		<a href="/support/guides/tool-guide/sections/gitools-GS-capabilities">Gitools &gt;&gt;</a>
</p>
</td></tr>
<script type="text/javascript">
//<![CDATA[

		if ( "Tool Guide" == "Recipes" && true ) {
			$.each($("#main_content a.toggle_link"), function() {
				$(this).click();
				var image = $(this).children("img");
				$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
			});
		}
		$('.collapsed').toggle();

//]]>
</script>
</tr></table>
</div>
</div>
<div id="footer">
</div>
<script type="text/javascript">
//<![CDATA[

	$(document).ready(function() {
		$("a.toggle_link").click(function() {
			var image = $(this).children("img");
			$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
		});
	});

//]]>
</script>
<script type="text/javascript">
//<![CDATA[

  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54489169-1', 'auto');
  ga('send', 'pageview');


//]]>
</script></div></body>
</html>
