<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">

<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
<title>GenomeSpace: 
	  	Tool Guide
  </title>
<link href="/assets/site.css" media="screen" rel="stylesheet" type="text/css"/>
<link href="/assets/demo_table_jui.css" media="screen" rel="stylesheet" type="text/css"/>
<link href="/assets/jquery-ui-1.8.4.custom.css" media="screen" rel="stylesheet" type="text/css"/>
<script src="/assets/application.js" type="text/javascript"></script>
<link href="/assets/favicon.ico" rel="shortcut icon" type="image/vnd.microsoft.icon"/>
<meta content="authenticity_token" name="csrf-param"/>
<meta content="KFYLIt5be2eitmDM6PkCDj1Mrm0axPg15h0wm30QLaI=" name="csrf-token"/>
</head>
<body>
<div id="page_wrapper">
<div id="page_content">
<div id="logo"><a href="/"><img alt="Gslogo" src="/assets/gslogo.png"/></a></div>
<div id="login_menu"><a href="https://gsui.genomespace.org/jsui/gsui.html">User Login</a>
</div>
<ul id="nav">
<li id="what-is-genomespace"><a href="/what-is-genomespace">What is GenomeSpace?</a>
</li>
<li id="tools"><a href="/support/tools">Tools</a>
</li>
<li id="recipes"><a href="http://recipes.genomespace.org">Recipes</a></li>
<li id="documentation"><a href="/documentation">Documentation</a>
<ul>
<li><a href="/support/guides/user-guide">User Guide</a>
<p>Read the detailed guide to the user interface</p>
</li>
<li><a href="/support/guides/tool-guide">Tool Guide</a>
<p>Learn how to access GenomeSpace functionality for each Tool</p>
</li>
<li><a href="/faq">FAQ</a>
<p>See answers to frequently asked questions</p>
</li>
<li><a href="/presentations-and-videos">Presentations and Videos</a>
<p>Access talks by the GenomeSpace community</p>
</li>
<li><a href="/support/guides/end-to-end-analyses">End-to-End Analyses</a>
<p>Provides description for end-to-end analyses.</p>
</li>
</ul>
</li>
<li id="developers"><a href="/for-developers">Developers</a>
<ul>
<li><a href="/what-can-genomespace-do-for-tool-developers">What Can GenomeSpace Do For Tool Developers?</a>
<p></p>
</li>
<li><a href="/adopters">Technical Documentation</a>
<p></p>
</li>
<li><a href="/source-code">Source Code and Distributions</a>
<p></p>
</li>
<li><a href="/developer-support-and-feedback">Developer Support and Feedback</a>
<p></p>
</li>
</ul>
</li>
<li id="support"><a href="/support">Support</a>
</li>
<li id="about"><a href="/about">About</a>
</li>
<li class="search">
<div id="search_box">
<form accept-charset="UTF-8" action="/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="✓"/></div>
<input class="search" id="search" name="search" type="text"/>
<input class="searchBtn" type="submit"/>
</form> </div>
</li>
</ul>
<script type="text/javascript">
//<![CDATA[

		// grab page category and use to highlight correct menu item.  if none present, parse url to find where the user is.
		var category = "";
		if ( !category ){
			url_parts = $(location).attr('pathname').split("/");
			url_string = url_parts[1] + "/" + url_parts[2]
			if (url_string == "support/tools"){
				category = "tools"
			}
			else if (url_string == "support/guides"){
				url_parts[3] == "recipes" ? category = "recipes" : category = "documentation"
			}
			else if (url_parts[1] == "genomespace-recipes")
			{
                category = "gsrecipes"
            }
			else if (url_string == "support/api"){
				category = "developers"
			}
			else if ((url_string == "register/undefined") || (url_string == "request_invite/undefined")) {
				category = "what-is-genomespace"
			}
			else if (url_string == "/undefined") {
				category = ""
			}
			else if (url_string == "support/undefined") {
				category = "support"
			}
		}
		$("#" + category).addClass("current");


//]]>
</script>
<div id="content_panel">
<table width="100%"><tr>
<td id="toc_wrapper">
<div id="toc">
<h3>Tool Guide</h3>
<ol>
<li class="" id="guide_section_1">
<a href="/support/guides/tool-guide/sections/tool-guide-intro">Introduction</a>
</li>
<li class="" id="guide_section_2">
<a href="/support/guides/tool-guide/sections/arrayexpress-GS-capabilities">ArrayExpress</a>
</li>
<li class="" id="guide_section_3">
<a href="/support/guides/tool-guide/sections/cbioportal-GS-capabilities">cBioPortal</a>
</li>
<li class="" id="guide_section_4">
<a href="/support/guides/tool-guide/sections/cistrome-GS-capabilities">Cistrome </a>
</li>
<li class="" id="guide_section_5">
<a href="/support/guides/tool-guide/sections/cytoscape-GS-capabilities">Cytoscape</a>
</li>
<li class="" id="guide_section_6">
<a href="/support/guides/tool-guide/sections/cytoscape_3-GS-capabilities">Cytoscape 3</a>
</li>
<li class="" id="guide_section_7">
<a href="/support/guides/tool-guide/sections/firebrowse-GS-capabilities">FireBrowse</a>
</li>
<li class="" id="guide_section_8">
<a href="/support/guides/tool-guide/sections/galaxy-GS-capabilities">Galaxy</a>
</li>
<li class="" id="guide_section_9">
<a href="/support/guides/tool-guide/sections/genepattern-GS-capabilities">GenePattern</a>
</li>
<li class="" id="guide_section_10">
<a href="/support/guides/tool-guide/sections/genomica-GS-capabilities">Genomica</a>
</li>
<li class="" id="guide_section_11">
<a href="/support/guides/tool-guide/sections/geworkbench-GS-capabilities">geWorkbench</a>
</li>
<li class="" id="guide_section_12">
<a href="/support/guides/tool-guide/sections/gitools-GS-capabilities">Gitools</a>
</li>
<li class="" id="guide_section_13">
<a href="/support/guides/tool-guide/sections/igv-gs-capabilities">Integrative Genomics Viewer (IGV)</a>
</li>
<li class="" id="guide_section_14">
<a href="/support/guides/tool-guide/sections/isacreator">ISAcreator</a>
</li>
<li class="" id="guide_section_15">
<a href="/support/guides/tool-guide/sections/msigdb-online-tools">MSigDB Online Tools</a>
</li>
<li class="" id="guide_section_16">
<a href="/support/guides/tool-guide/sections/ucsc_genome_browser-GS-capabilities">UCSC Table Browser</a>
</li>
<li class="" id="guide_section_17">
<a href="/support/guides/tool-guide/sections/tool-term-glossary">Tool Term Glossary</a>
</li>
<li class="" id="guide_section_18">
<a href="/support/guides/tool-guide/sections/trinity-GS-capabilities">Trinity</a>
</li>
</ol>
</div>
<p class="center"><a href="/support/guides/tool-guide/print"><img alt="Print_icon" src="/assets/print_icon.gif" title="Printer Friendly View"/></a></p>
</td>
<script type="text/javascript">
//<![CDATA[

		$(document).ready(function() {
			var guide_section = "guide_section_5";
			$("#" + guide_section + "> a").css("font-weight", "bold");
			if ( 'Tool Guide' == "Recipes" ) {
				$.each($("#toc a.toggle_link"), function() {
					$(this).click();
					var image = $(this).children("img");
					$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
				});
			}
		});

//]]>
</script>
<td id="main_content">
<div id="notice"></div>
<div id="alert"></div>
<h1>Cytoscape
	</h1>
<p>
	Project Website: <a href="http://www.cytoscape.org">http://www.cytoscape.org</a></p>
<p>
	Cytoscape is an open-source bioinformatics software platform for visualizing molecular interaction networks and biological pathways, and integrating these networks with annotations, gene expression profiles, and other state data.   Cytoscape core distribution provides a basic set of features for data integration and visualization.  Additional features are available as plugins.</p>
<p>
	Cytoscape is a Java-based application that downloads to your local machine when you launch it from GenomeSpace.  Once it's downloaded to your machine (after a launch from GenomeSpace), you can start it locally (by opening the JNLP) or launch from GenomeSpace.  If you launch Cytoscape from the Cytoscape website, at this time it will not be the version that is linked to GenomeSpace.</p>
<p>
	Like several other applications in the GenomeSpace suite of tools, Cytoscape can be expanded with plug-ins.  You can download plug-ins for your Cytoscape by choosing <em>Plugins&gt;Manage Plugins</em>.</p>
<p>
	Cytoscape has several GenomeSpace menu items.</p>
<h2>
	Logging In</h2>
<p>
	When you launch Cytoscape from GenomeSpace, your login should be handled seamlessly.</p>
<p>
	If you start from your GenomeSpace-enabled JNLP of Cytoscape and want to log into GenomeSpace in order to access your Amazon cloud storage or other GenomeSpace tools, you will be asked to log in with your GenomeSpace username and password.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytogslogin.jpg" style="width: 300px; height: 206px;"/></p>
<p>
	You can also <em>Reset Password</em> on your existing GenomeSpace account.</p>
<h2>
	Import Menu</h2>
<p>
	Clicking <em>File&gt;Import&gt;GenomeSpace</em> provides several options for importing files from GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoimportgs.jpg" style="width: 600px; height: 389px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 597px; height: 164px;">
<tbody>
<tr>
<td>
				Load Network</td>
<td>
				Loads a network from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Load Network from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of a network from a delimited text file or single-sheet MS Excel workbook.</td>
</tr>
<tr>
<td>
				Load Attributes</td>
<td>
				Loads attributes (for node/edge/network) from your GenomeSpace cloud storage. The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load CyTable</td>
<td>
				Loads a file containing attributes for a CyNetwork from your GenomeSpace cloud storage.  The load dialog allows you to opt to either <em>Load node attributes</em> or <em>Load edge attributes</em>.</td>
</tr>
<tr>
<td>
				Load Attributes from Table</td>
<td>
				Starts the import from your GenomeSpace cloud storage of delimited text or single-sheet MS Excel workbook containing attribute data tables that are not formatted into Cytoscape node or edge attribute file formats.</td>
</tr>
<tr>
<td>
				Load Session</td>
<td>
				Loads a CYS file from your GenomeSpace cloud storage.  A session file (CYS) contains all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles present in a given working state.</td>
</tr>
<tr>
<td>
				Load Ontology and Annotations</td>
<td>
				Loads an annotation/ontology file from your GenomeSpace cloud storage. Annotations in Cytoscape are stored as a set of ontologies (a set of controlled vocabulary terms that annotate the objects).The standard file formats used in Cytoscape are OBO and Gene Association.</td>
</tr>
</tbody>
</table>
<h2>
<br/>
	Export Menu</h2>
<p>
	Clicking <em>File&gt;Export&gt;GenomeSpace</em> provides options for exporting files to GenomeSpace.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytoexportgs.jpg" style="width: 600px; height: 353px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 598px; height: 126px;">
<tbody>
<tr>
<td>
				Upload File</td>
<td>
				Uploads a file from your local machine to the root level of your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Session As</td>
<td>
				Saves a session file (CYS), containing all networks, attributes (for node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles in your present working state to your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Save Network As</td>
<td>
				Save a network to your GenomeSpace cloud storage.</td>
</tr>
</tbody>
</table>
<h2>
	GenomeSpace Menu</h2>
<p>
	Clicking <em>File&gt;GenomeSpace</em> provides several options for managing your files in your GenomeSpace cloud storage and using GenomeSpace to launch other applications.</p>
<p>
<img alt="" src="/uploaded/content_toolug_cytofilegs.jpg" style="width: 600px; height: 365px;"/></p>
<table border="1" cellpadding="1" cellspacing="1" style="width: 500px;">
<tbody>
<tr>
<td>
				Delete File in GenomeSpace</td>
<td>
				Delete a file from your GenomeSpace cloud storage.</td>
</tr>
<tr>
<td>
				Download File</td>
<td>
				Download a file in your GenomeSpace cloud storage to your local machine.</td>
</tr>
<tr>
<td>
				List Available Files in GenomeSpace</td>
<td>
				Navigate your GenomeSpace directories.</td>
</tr>
<tr>
<td>
				Login To GenomeSpace</td>
<td>
				Log into your GenomeSpace account.</td>
</tr>
<tr>
<td>
				Launch</td>
<td>
				Launch another GenomeSpace tool</td>
</tr>
</tbody>
</table>
<p>
	 </p>
</td>
<tr><td colspan="2">
<p class="center">
<a href="/support/guides/tool-guide/sections/cistrome-GS-capabilities">&lt;&lt; Cistrome </a>
		| <a href="/support/guides/tool-guide/sections/tool-guide-intro">Up</a> |
		<a href="/support/guides/tool-guide/sections/cytoscape_3-GS-capabilities">Cytoscape 3 &gt;&gt;</a>
</p>
</td></tr>
<script type="text/javascript">
//<![CDATA[

		if ( "Tool Guide" == "Recipes" && true ) {
			$.each($("#main_content a.toggle_link"), function() {
				$(this).click();
				var image = $(this).children("img");
				$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
			});
		}
		$('.collapsed').toggle();

//]]>
</script>
</tr></table>
</div>
</div>
<div id="footer">
</div>
<script type="text/javascript">
//<![CDATA[

	$(document).ready(function() {
		$("a.toggle_link").click(function() {
			var image = $(this).children("img");
			$(image).attr("src", $(image).attr("src") == "/assets/plus.png" ? "/assets/minus.png" : "/assets/plus.png");
		});
	});

//]]>
</script>
<script type="text/javascript">
//<![CDATA[

  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54489169-1', 'auto');
  ga('send', 'pageview');


//]]>
</script></div></body>
</html>
